app/services/notify_extra_mobile_data_account_holder_service.rb

```diff
@@ -0,0 +1,56 @@
+class NotifyExtraMobileDataAccountHolderService
```

Comment:
(The following suggestion is just personal taste and comes from a general dislike of service objects, so take with a pinch of 🧂 )

This class could be made more object-oriented by renaming it to something like `ExtraMobileDataAccountHolderNotification`.  A `deliver!` method on that class would do the sending. You could construct it off the `ExtraMobileDataRequest` class:

```
extra_mobile_data_request.acount_holder_notification.deliver!
```

Reply:
I second that 👍

Reply:
Not implemented it this as yet, although I do like it. Wanted to get it all working and inline with the rubocop rules as they are setup.

Reply:
Perhaps we should also consider backgrounding the call to Notify once @aldavidson has established the approach

Reply:
yep, totally -  but don't let that hold up this PR. Once my PR has been merged, then we can do separate smaller PRs for porting individual tasks over to jobs